kolibri/plugins/coach/assets/src/composables/useFetchContentNode.js

```diff
@@ -0,0 +1,38 @@
+import { ref } from 'vue';
+import ContentNodeResource from 'kolibri-common/apiResources/ContentNodeResource';
+
+export default function useFetchContentNode(contentId) {
+  const contentNode = ref({});
+  const ancestors = ref([]);
+  const questions = ref([]);
+  const loading = ref(true);
+
+  const fetchContentNode = async () => {
+    ContentNodeResource.fetchModel({
+      id: contentId,
+      getParams: { no_available_filtering: true },
+    }).then(node => {
+      contentNode.value = node;
+      if (node != null) {
+        loading.value = false;
```

Comment:
If it is null then will loading be true indefinitely?

Reply:
Yes,  I see!  if `node `is null, the` loading.value `will remain true indefinitely because the condition to set` loading.value = false `is only executed when` node `is not `null. ` I can can see an infinite loading state here. Thanks for pointing that out

Reply:
I'm not sure `node` can ever be `null` here - I don't think our [api resource methods](https://github.com/learningequality/kolibri/blob/072f8eeb57754705d03ca02afe01d672b7825b74/packages/kolibri/apiResource.js#L739) like `fetchModel` would return anything besides the data it gets from the API request _or_ it throws an error.

So, in this case, I think you'd probably want to just treat your `then()` callback function as if it's exactly what you expected to get.

Then you can chain a catch method on it _in the component where you call `fetchContentNode`_ that way you can handle the errors there in the Vue component however best works there.